src/utils/RunSUSHI.js

```diff
@@ -11,6 +13,73 @@ const getRandomPun = utils.getRandomPun;
 const Type = utils.Type;
 const FHIRDefinitions = fhirdefs.FHIRDefinitions;
 
+async function loadAndCleanDatabase(defs, dependencies) {
+  let helperUpdate = await checkForDatabaseUpgrade(dependencies);
+  let loadExternalDependenciesReturn = { defs, emptyDependencies: [] };
+
+  if (helperUpdate.shouldUpdate) {
+    loadExternalDependenciesReturn = await loadExternalDependencies(defs, helperUpdate.version + 1, dependencies);
+    defs = loadExternalDependenciesReturn.defs;
+  } else {
+    loadExternalDependenciesReturn = await loadExternalDependencies(defs, helperUpdate.version, dependencies);
+    defs = loadExternalDependenciesReturn.defs;
+  }
+
+  // Cleans out database of any empty objectStores
+  await cleanDatabase(loadExternalDependenciesReturn.emptyDependencies, helperUpdate.version + 2);
+  return defs;
+}
+
+/**
+ * Run GoFSH
+ * @param {array} input array of JSON definitions to be processed
+ * @param {object} options - config options for GoFSH based on user input and defaults
+ * dependencies: user set, defaults to []
+ * @param {FHIRDefinitions} testDefs - this should only be used by the unit tests so they can provide their own definitions.
+ * @returns {string} the FSH
+ */
+export async function runGoFSH(input, options, testDefs = null) {
+  // Read in the resources, either as JSON objects or as strings
+  const docs = []; // WildFHIR[]
+  input.forEach((resource, i) => {
+    const location = `Input_${i}`;
+    if (typeof resource === 'string') {
```

Comment:
Can the input be anything other than strings here?

Reply:
Hm good point! This was really just here because I used so much of `FhirToFsh`, but since I'm not actually using that, I can simplify this for now and just `JSON.parse` each input resource.

Reply:
Okay actually it turns out that the test FSH that I have committed here is actually objects. I don't think the text we get from the editor will actually be objects, I think it will be strings, but do you think it's okay to keep this for now for our testing purposes and for alignment with `FhirToFsh`? Or do you think it's preferable to just stringify the temporary testing objects I have now?

Reply:
I've stringified all the things! b2aa777